sqlitecluster/SQLiteNode.cpp

```diff
@@ -1655,6 +1655,15 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
                             }
                         }
                     }
+                    const int64_t currentCommitDifference = message.calcU64("NewCount") - getCommitCount();
+                    const string blockReason = "COMMITS_LAGGING_BEHIND";
+                    if (currentCommitDifference > 50'000 && !_server.isCommandPortClosed(blockReason)) {
+                        SINFO("Node is lagging behind, closing command port so it can catch up.");
+                        _server.blockCommandPort(blockReason);
+                    } else if (currentCommitDifference < 10'000 && _server.isCommandPortClosed(blockReason)) {
+                        SINFO("Node is caught up enough, unblocking command port.");
+                        _server.unblockCommandPort(blockReason);
+                    }
```

Comment:
Is there a reason this block is here in particular? We have this data everywhere in this method, we don't need to do this only after starting a replicate thread (and it seems like we'd rather do it before starting a replicate thread, though I doubt it makes a practical difference).